src/patternfly/components/InputGroup/styles.scss

```diff
@@ -29,7 +29,8 @@
     margin-left: -1px;
   }
   /* stylelint-enable */
-  .pf-m-invalid {
+
+  .pf-c-form-control[aria-invalid="true"] {
     margin-left: 0;
```

Comment:
I think it looks better without this and the red border is on top of the grey border :)